server/analytic.go

```diff
@@ -1,64 +0,0 @@
-package main
-
-import (
-	"sync"
-	"time"
-)
-
-type Analytic struct {
-	lock          sync.RWMutex
-	Start         time.Time
-	End           time.Time
-	Channels      map[string]int64
-	ChannelsReply map[string]int64
-	Users         map[string]int64
-	UsersReply    map[string]int64
-	FilesNb       int64
-	FilesSize     int64
-}
-
-func NewAnalytic() *Analytic {
-	return &Analytic{
-		lock:          sync.RWMutex{},
-		Start:         time.Now(),
-		Channels:      make(map[string]int64),
-		ChannelsReply: make(map[string]int64),
-		Users:         make(map[string]int64),
-		UsersReply:    make(map[string]int64),
-		FilesNb:       int64(0),
-		FilesSize:     int64(0),
-	}
-}
-
-func (a *Analytic) Init() {
```

Comment:
Why is this needed if you have `NewAnalytic`?

Reply:
Good catch :D It is a hack because when I create a new session I hold a reference to the currentAnalytic. I need to invistage after all PR are merged https://github.com/manland/mattermost-plugin-analytics/issues/17